src/base/http/connection.h

```diff
@@ -27,41 +28,69 @@
  * exception statement from your version.
  */
 
-
 #pragma once
 
 #include <QElapsedTimer>
 #include <QObject>
 
+#include "types.h"
+
 class QTcpSocket;
 
 namespace Http
 {
-    class IRequestHandler;
-    struct Response;
-
     class Connection : public QObject
     {
         Q_OBJECT
         Q_DISABLE_COPY(Connection)
 
     public:
-        Connection(QTcpSocket *socket, IRequestHandler *requestHandler, QObject *parent = nullptr);
-        ~Connection();
+        enum Status
+        {
+            PendingRequest,
+            PendingStatus,
+            PendingHeaders,
+            PendingContent,
+            WritingContent,
+            Closed
+        };
+
+        // Connection takes ownership of socket
+        explicit Connection(QTcpSocket *socket, QObject *parent = nullptr);
 
-        bool hasExpired(qint64 timeout) const;
-        bool isClosed() const;
+        void sendStatus(const ResponseStatus &status);
+        void sendHeaders(const HeaderMap &headers);
+        void sendContent(const QByteArray &data);
```

Comment:
Also you can already different data (status, content, etc.) by the function signature, you can use function overload and have the same function name for all.

Reply:
I find overloads for such non trivial actions really confusing especially since there will be three of them.